flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskExecutionDecorationTest.java

```diff
@@ -0,0 +1,172 @@
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Verifies that {@link StreamTask} {@link StreamTaskActionExecutor decorates execution} of actions that potentially needs to be synchronized.
+ */
+public class StreamTaskExecutionDecorationTest {
+	private CountingStreamTaskActionExecutor decorator;
+	private StreamTask<Object, StreamOperator<Object>> task;
+
+	@Test
+	public void testAbortCheckpointOnBarrierIsDecorated() throws Exception {
+		task.abortCheckpointOnBarrier(1, null);
+		verify();
+	}
+
+	@Test
+	public void testTriggerCheckpointOnBarrierIsDecorated() throws Exception {
+		task.triggerCheckpointOnBarrier(new CheckpointMetaData(1, 2), new CheckpointOptions(CheckpointType.CHECKPOINT, new CheckpointStorageLocationReference(new byte[]{1})), null);
+		verify();
+	}
+
+	@Test
+	public void testTriggerCheckpointAsyncIsDecorated() throws Exception {
+		task.triggerCheckpointAsync(new CheckpointMetaData(1, 2), new CheckpointOptions(CheckpointType.CHECKPOINT, new CheckpointStorageLocationReference(new byte[]{1})), false);
+		new Thread(() -> {
+			try {
+				Thread.sleep(50);
+			} catch (InterruptedException ex) {
+				ex.printStackTrace();
+			} finally {
+				task.mailboxProcessor.allActionsCompleted();
+			}
+		}).start();
+		task.mailboxProcessor.runMailboxLoop();
+		verify();
+	}
+
+	@Test
+	public void testMailboxExecutorIsDecorated() throws Exception {
+		CountDownLatch latch = new CountDownLatch(1);
+		task.mailboxProcessor.getMainMailboxExecutor().asExecutor("test").execute(() -> {
+			try {
+				verify();
+			} finally {
+				latch.countDown();
+			}
+		});
+		new Thread(() -> {
+			try {
+				latch.await();
+				task.mailboxProcessor.allActionsCompleted();
+			} catch (Exception ex) {
+				ex.printStackTrace();
+			}
+		}).start();
+		task.mailboxProcessor.runMailboxLoop();
+	}
+
+	@Before
+	public void before() {
+		decorator = new CountingStreamTaskActionExecutor();
+		task = new StreamTask<Object, StreamOperator<Object>>(new StreamTaskTest.DeclineDummyEnvironment(), null, FatalExitExceptionHandler.INSTANCE, decorator) {
+			@Override
+			protected void init() {
+			}
+
+			@Override
+			protected void processInput(MailboxDefaultAction.Controller controller) {
+			}
+		};
+	}
+
+	private void verify() {
+		Assert.assertTrue("execution decorator was not called", decorator.wasCalled());
+	}
+
+	@After
+	public void after() {
+		decorator = null;
+		task = null;
+	}
+
+	@SuppressWarnings("CatchMayIgnoreException")
+	static class CountingStreamTaskActionExecutor extends StreamTaskActionExecutor.SynchronizedStreamTaskActionExecutor {
+		private AtomicInteger calls = new AtomicInteger(0);
+
+		CountingStreamTaskActionExecutor() {
+			super(new Object());
+		}
+
+		int getCallCount() {
+			return calls.get();
+		}
+
+		boolean wasCalled() {
+			return getCallCount() > 0;
+		}
+
+		@Override
+		public void run(Runnable runnable) {
+			calls.incrementAndGet();
+			try {
+				runnable.run();
+			} catch (Exception e) {
```

Comment:
Why are you catching and swallowing all the exceptions?

Reply:
The test only verifies that the executor was called regardless of the returned results or exceptions.

Reply:
I think it would be better to not catch those exceptions at all or propagate them as `RuntimeException`.

There can be some unexpected exception, for example NPE or OOM, that indicates a bug in the test. Maybe the test will fail anyway later, but exception swallowing will make debugging more difficult since it won't be reported and may leave the code in an inconsistent state.

Reply:
OOM (or any other VM-level error) isn't caught by this block, and NPE (or any other Task-level exception) is fine here as Task isn't fully initialized.

Reply:
Addressed by proper task initialization.